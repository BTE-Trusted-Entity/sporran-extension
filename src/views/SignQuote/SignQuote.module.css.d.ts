export const __esModule: true;
export const container: string;
export const heading: string;
export const subline: string;
export const details: string;
export const detailName: string;
export const detailValue: string;
export const costs: string;
export const label: string;
export const buttonsLine: string;
export const cancel: string;
export const submit: string;
export const errorTooltip: string;
