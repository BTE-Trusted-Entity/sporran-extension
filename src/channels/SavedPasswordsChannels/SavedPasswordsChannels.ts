import { BrowserChannel } from '../base/BrowserChannel/BrowserChannel';
import { channelsEnum } from '../base/channelsEnum';

interface SavedPassword {
  password: string;
  timestamp: number;
}

const savedPasswords: Record<string, SavedPassword> = {};

interface SavePasswordInput {
  password: string;
  address: string;
}

export const savePasswordChannel = new BrowserChannel<SavePasswordInput>(
  channelsEnum.savePassword,
);

export async function savePassword({
  address,
  password,
}: SavePasswordInput): Promise<void> {
  savedPasswords[address] = { password, timestamp: Date.now() };
}

export const getPasswordChannel = new BrowserChannel<
  string,
  string | undefined
>(channelsEnum.getPassword);

export async function getPassword(
  address: string,
): Promise<string | undefined> {
  return savedPasswords[address]?.password;
}

export const forgetPasswordChannel = new BrowserChannel<string>(
  channelsEnum.forgetPassword,
);

export async function forgetPassword(address: string): Promise<void> {
  delete savedPasswords[address];
}

export const hasSavedPasswordsChannel = new BrowserChannel<void, boolean>(
  channelsEnum.hasSavedPasswords,
);

export async function hasSavedPasswords(): Promise<boolean> {
  return Object.values(savedPasswords).length > 0;
}

export const forgetAllPasswordsChannel = new BrowserChannel(
  channelsEnum.forgetAllPasswords,
);

export async function forgetAllPasswords(): Promise<void> {
  for (const password in savedPasswords) {
    delete savedPasswords[password];
  }
}

const saveDuration = 15 * 60 * 1000;
const intervalDuration = 1 * 60 * 1000;

function checkExpiredPasswords(): void {
  const oldestPossible = Date.now() - saveDuration;
  for (const password in savedPasswords) {
    const { timestamp } = savedPasswords[password];
    if (timestamp < oldestPossible) {
      delete savedPasswords[password];
    }
  }
}

export function schedulePasswordsCheck(): void {
  setInterval(checkExpiredPasswords, intervalDuration);
}
