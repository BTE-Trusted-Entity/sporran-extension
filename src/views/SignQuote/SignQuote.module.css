.container {
  composes: container from '../../components/View/View.module.css';
}

.heading {
  composes: screenHeadline from '../../components/Typography/Typography.module.css';
}

.subline {
  composes: subline from '../../components/Typography/Typography.module.css';
  margin: 0 0 0.8125rem;
}

.details {
  composes: subline from '../../components/Typography/Typography.module.css';
  margin: 0;
  display: flex;
  justify-content: flex-end;
  flex-wrap: wrap;
}

.detailName {
  width: 30%;
  margin: 0.125rem;
  text-align: left;
}

.detailValue {
  width: 50%;
  margin: 0.125rem;
  text-align: left;
}

.costs {
  composes: smallText from '../../components/Typography/Typography.module.css';
  font-size: 1.25rem;
  margin: 0.5rem 0 1rem;
}

.label {
  composes: smallText from '../../components/Typography/Typography.module.css';
  text-align: left;
}

.buttonsLine {
  composes: buttonsLine from '../../components/Button/Button.module.css';
  margin: 0;
}

.cancel {
  composes: buttonSecondary from '../../components/Button/Button.module.css';
}

.submit {
  composes: buttonPrimary from '../../components/Button/Button.module.css';
}

.errorTooltip {
  composes: tooltip from '../../components/ErrorTooltip/ErrorTooltip.module.css';
  top: 2.5rem;
}
