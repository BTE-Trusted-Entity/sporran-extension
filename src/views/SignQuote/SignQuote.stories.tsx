import { Meta } from '@storybook/react';

import { mockTerms } from '../../utilities/mockTerms/mockTerms';

import { PopupTestProvider } from '../../utilities/popups/PopupTestProvider';
import { moreIdentitiesMock as identities } from '../../utilities/identities/IdentitiesProvider.mock';
import { paths } from '../paths';

import { SignQuote } from './SignQuote';

export default {
  title: 'Views/SignQuote',
  component: SignQuote,
} as Meta;

export function Template(): JSX.Element {
  return (
    <PopupTestProvider path={paths.popup.claim} data={mockTerms}>
      <SignQuote
        identity={
          identities['4tDjyLy2gESkLzvaLnpbn7N61VgnwAhqnTHsPPFAwaZjGwP1']
        }
      />
    </PopupTestProvider>
  );
}

export function OnChainDidDeleted(): JSX.Element {
  return (
    <PopupTestProvider path={paths.popup.claim} data={mockTerms}>
      <SignQuote
        identity={
          identities['4rZ7pGtvmLhAYesf7DAzLQixdTEwWPN3emKb44bKVXqSoTZB']
        }
      />
    </PopupTestProvider>
  );
}
