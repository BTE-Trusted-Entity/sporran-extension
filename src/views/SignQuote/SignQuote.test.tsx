import { CType } from '@kiltprotocol/sdk-js';

import {
  moreIdentitiesMock as identities,
  render,
} from '../../testing/testing';
import { mockTerms } from '../../utilities/mockTerms/mockTerms';
import { PopupTestProvider } from '../../utilities/popups/PopupTestProvider';
import { parseDidUri } from '../../utilities/did/did';
import { waitForGetPassword } from '../../channels/SavedPasswordsChannels/SavedPasswordsChannels.mock';
import { paths } from '../paths';

import { useIsOnChainDidDeleted } from '../../utilities/did/useIsOnChainDidDeleted';

import { SignQuote } from './SignQuote';

jest.mocked(CType.hashToId).mockImplementation((id) => `kilt:ctype:${id}`);

jest.mock('../../utilities/did/did');
jest.mocked(parseDidUri).mockReturnValue({
  fullDid: 'did:kilt:4rrkiRTZgsgxjJDFkLsivqqKTqdUTuxKk3FX3mKFAeMxsR51',
} as unknown as ReturnType<typeof parseDidUri>);

jest.mock('../../utilities/did/useIsOnChainDidDeleted');

describe('SignQuote', () => {
  it('should render', async () => {
    jest.mocked(useIsOnChainDidDeleted).mockReturnValue(false);

    const { container } = render(
      <PopupTestProvider path={paths.popup.claim} data={mockTerms}>
        <SignQuote
          identity={
            identities['4tDjyLy2gESkLzvaLnpbn7N61VgnwAhqnTHsPPFAwaZjGwP1']
          }
        />
      </PopupTestProvider>,
    );
    await waitForGetPassword();
    expect(container).toMatchSnapshot();
  });

  it('should show error message if on-chain DID deleted', async () => {
    jest.mocked(useIsOnChainDidDeleted).mockReturnValue(true);

    const { container } = render(
      <PopupTestProvider path={paths.popup.claim} data={mockTerms}>
        <SignQuote
          identity={
            identities['4rZ7pGtvmLhAYesf7DAzLQixdTEwWPN3emKb44bKVXqSoTZB']
          }
        />
      </PopupTestProvider>,
    );
    await waitForGetPassword();
    expect(container).toMatchSnapshot();
  });
});
