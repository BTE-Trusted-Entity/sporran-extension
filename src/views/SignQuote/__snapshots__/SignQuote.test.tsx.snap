// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignQuote should render 1`] = `
<div>
  <form
    autocomplete="off"
    class="container"
  >
    <h1
      class="heading"
    >
      Terms of Attestation
    </h1>
    <p
      class="subline"
    >
      The credential will be issued to this Identity
    </p>
    <section
      class="container"
    >
      <div
        class="avatar"
      >
        Identicon
      </div>
      <div
        class="centeredNameLine"
      >
        <span
          class="name"
        >
          Light DID Identity
        </span>
      </div>
    </section>
    <p
      class="costs"
    >
      <span>
        Credential Costs: 
      </span>
      <span
        class="amount"
      >
        233
        <span
          class="fractional"
        >
          .0000
        </span>
         
        <span
          aria-label="KILT Coin"
          class="coin"
        />
      </span>
    </p>
    <dl
      class="details"
    >
      <dt
        class="detailName"
      >
        Full Name
        :
      </dt>
      <dd
        class="detailValue"
      >
        Ingo Rï¿½be
      </dd>
      <dt
        class="detailName"
      >
        Email
        :
      </dt>
      <dd
        class="detailValue"
      >
        ingo@kilt.io
      </dd>
      <dt
        class="detailName"
      >
        Credential type
        :
      </dt>
      <dd
        class="detailValue"
      >
        BL-Mail-Simple
      </dd>
      <dt
        class="detailName"
      >
        Attester
        :
      </dt>
      <dd
        class="detailValue"
      >
        SocialKYC
      </dd>
    </dl>
    <p
      class="resetLine"
    >
      <label
        class="passwordLabel"
        for="password"
      >
        Sign with password:
      </label>
      <a
        class="reset"
        href="/identity/4tDjyLy2gESkLzvaLnpbn7N61VgnwAhqnTHsPPFAwaZjGwP1/reset"
      >
        Reset password
      </a>
    </p>
    <p
      class="passwordLine"
    >
      <input
        class="password"
        id="password"
        name="password"
        type="password"
        value=""
      />
      <button
        aria-label="Show"
        class="show"
        title="Show"
        type="button"
      />
      <output
        class="errorTooltip"
        hidden=""
      />
    </p>
    <label
      class="rememberLabel"
    >
      <span>
        Remember password for the next 15 minutes
      </span>
      <input
        class="remember"
        name="remember"
        type="checkbox"
      />
      <span />
    </label>
    <p
      class="buttonsLine"
    >
      <button
        class="cancel"
        type="button"
      >
        Cancel
      </button>
      <button
        class="submit"
        type="submit"
      >
        Request Attestation
      </button>
      <output
        class="errorTooltip"
        hidden=""
      >
        DID deleted, please select a different Identity
      </output>
    </p>
  </form>
</div>
`;

exports[`SignQuote should show error message if on-chain DID deleted 1`] = `
<div>
  <form
    autocomplete="off"
    class="container"
  >
    <h1
      class="heading"
    >
      Terms of Attestation
    </h1>
    <p
      class="subline"
    >
      Choose an Identity for your credential
    </p>
    <div
      class="container"
    >
      <section
        class="container"
      >
        <div
          class="avatar"
        >
          Identicon
        </div>
        <div
          class="centeredNameLine"
        >
          <span
            class="name"
          >
            Identity with deleted on-chain DID
          </span>
        </div>
      </section>
      <a
        aria-label="Add Identity"
        class="left"
        href="/?data=eyJjbGFpbSI6eyJjVHlwZUhhc2giOiIweDI0MGM3NDQ5MjNiZDk4Nzk3NTA0MzI4ZjliZWY1N2NhNmM3NzdmNWNiZjNlMWFhZGE3NGYzNDhjOTg3OWQ3OGEiLCJjb250ZW50cyI6eyJGdWxsIE5hbWUiOiJJbmdvIFL8YmUiLCJFbWFpbCI6ImluZ29Aa2lsdC5pbyJ9LCJvd25lciI6ImRpZDpraWx0OmxpZ2h0OjAwNG9VaUszRXZUY3pjNnVrWUxFVjU3QnV4emtXOXprc0tSY2haWFlHU29jTG1XRTVOIn0sImxlZ2l0aW1hdGlvbnMiOltdLCJxdW90ZSI6eyJhdHRlc3RlclNpZ25hdHVyZSI6eyJrZXlVcmkiOiJkaWQ6a2lsdDo0b2VKNzZoZGo4NHhud0NOcWlqVUhVQ1RtZndYZ1NaNHZteExFaVRFWWdRZEJDY1ojMHg1OWE1NmUxYTQwMTdiNjBhMDBjNTQ1MGJmZDZhZjIxZGM3NTNlODRlZjcyZjJmZDQ1ZGZjMjdkZDVjM2JkYWZiIiwic2lnbmF0dXJlIjoiMHgzMGQwMWFlZmQ2NGM3ZWVhNDhiYzc0Mzc4MmZlNTRiYTU4Zjk2YjVhOTFlMzNkOGQ5ZWU1ZmFlMjgyOTY0OTY3YmRlYTVlOWFjM2YzMTc1NmZkMjUxYTQwZWE0ZDJlOWE5M2FjOWIxMzE0Y2MwMzdjZGNiMDRhNTNhMDA3ZTY4ZiJ9LCJhdHRlc3RlckRpZCI6ImRpZDpraWx0OjRvZUo3Nmhkajg0eG53Q05xaWpVSFVDVG1md1hnU1o0dm14TEVpVEVZZ1FkQkNjWiIsImNUeXBlSGFzaCI6IjB4YmZhZDYwOTc3YmMxOGNmOWRmZDc2ZGE4ODYyNGNlMjE5MzYxZjMzN2I0MzMyZDVjNDJjMDQ3NDk5ZjRiOTNjNyIsImNvc3QiOnsiZ3Jvc3MiOjIzMywibmV0IjoyMy4zLCJ0YXgiOnsidmF0IjozLjN9fSwiY3VycmVuY3kiOiJLSUxUIiwidGltZWZyYW1lIjoiU2F0IEp1bCAxMCAyMDIxIDAwOjAwOjAwIEdNVCswMDAwIChDb29yZGluYXRlZCBVbml2ZXJzYWwgVGltZSkiLCJ0ZXJtc0FuZENvbmRpdGlvbnMiOiJodHRwczovL3d3dy5leGFtcGxlLmNvbS90ZXJtcyJ9LCJjVHlwZXMiOlt7IiRpZCI6ImtpbHQ6Y3R5cGU6MHgyNDBjNzQ0OTIzYmQ5ODc5NzUwNDMyOGY5YmVmNTdjYTZjNzc3ZjVjYmYzZTFhYWRhNzRmMzQ4Yzk4NzlkNzhhIiwiJHNjaGVtYSI6Imh0dHA6Ly9raWx0LXByb3RvY29sLm9yZy9kcmFmdC0wMS9jdHlwZSMiLCJ0aXRsZSI6IkJMLU1haWwtU2ltcGxlIiwicHJvcGVydGllcyI6eyJGdWxsIG5hbWUiOnsidHlwZSI6InN0cmluZyJ9LCJFbWFpbCI6eyJ0eXBlIjoic3RyaW5nIn19LCJ0eXBlIjoib2JqZWN0In1dLCJhdHRlc3Rlck5hbWUiOiJTb2NpYWxLWUMiLCJhdHRlc3RlckRpZCI6ImRpZDpraWx0OjRvZUo3Nmhkajg0eG53Q05xaWpVSFVDVG1md1hnU1o0dm14TEVpVEVZZ1FkQkNjWiIsInNwZWNWZXJzaW9uIjoiMy4wIn0="
        title="Add Identity"
      />
      <a
        aria-label="Light DID Identity"
        class="right"
        href="/?data=eyJjbGFpbSI6eyJjVHlwZUhhc2giOiIweDI0MGM3NDQ5MjNiZDk4Nzk3NTA0MzI4ZjliZWY1N2NhNmM3NzdmNWNiZjNlMWFhZGE3NGYzNDhjOTg3OWQ3OGEiLCJjb250ZW50cyI6eyJGdWxsIE5hbWUiOiJJbmdvIFL8YmUiLCJFbWFpbCI6ImluZ29Aa2lsdC5pbyJ9LCJvd25lciI6ImRpZDpraWx0OmxpZ2h0OjAwNG9VaUszRXZUY3pjNnVrWUxFVjU3QnV4emtXOXprc0tSY2haWFlHU29jTG1XRTVOIn0sImxlZ2l0aW1hdGlvbnMiOltdLCJxdW90ZSI6eyJhdHRlc3RlclNpZ25hdHVyZSI6eyJrZXlVcmkiOiJkaWQ6a2lsdDo0b2VKNzZoZGo4NHhud0NOcWlqVUhVQ1RtZndYZ1NaNHZteExFaVRFWWdRZEJDY1ojMHg1OWE1NmUxYTQwMTdiNjBhMDBjNTQ1MGJmZDZhZjIxZGM3NTNlODRlZjcyZjJmZDQ1ZGZjMjdkZDVjM2JkYWZiIiwic2lnbmF0dXJlIjoiMHgzMGQwMWFlZmQ2NGM3ZWVhNDhiYzc0Mzc4MmZlNTRiYTU4Zjk2YjVhOTFlMzNkOGQ5ZWU1ZmFlMjgyOTY0OTY3YmRlYTVlOWFjM2YzMTc1NmZkMjUxYTQwZWE0ZDJlOWE5M2FjOWIxMzE0Y2MwMzdjZGNiMDRhNTNhMDA3ZTY4ZiJ9LCJhdHRlc3RlckRpZCI6ImRpZDpraWx0OjRvZUo3Nmhkajg0eG53Q05xaWpVSFVDVG1md1hnU1o0dm14TEVpVEVZZ1FkQkNjWiIsImNUeXBlSGFzaCI6IjB4YmZhZDYwOTc3YmMxOGNmOWRmZDc2ZGE4ODYyNGNlMjE5MzYxZjMzN2I0MzMyZDVjNDJjMDQ3NDk5ZjRiOTNjNyIsImNvc3QiOnsiZ3Jvc3MiOjIzMywibmV0IjoyMy4zLCJ0YXgiOnsidmF0IjozLjN9fSwiY3VycmVuY3kiOiJLSUxUIiwidGltZWZyYW1lIjoiU2F0IEp1bCAxMCAyMDIxIDAwOjAwOjAwIEdNVCswMDAwIChDb29yZGluYXRlZCBVbml2ZXJzYWwgVGltZSkiLCJ0ZXJtc0FuZENvbmRpdGlvbnMiOiJodHRwczovL3d3dy5leGFtcGxlLmNvbS90ZXJtcyJ9LCJjVHlwZXMiOlt7IiRpZCI6ImtpbHQ6Y3R5cGU6MHgyNDBjNzQ0OTIzYmQ5ODc5NzUwNDMyOGY5YmVmNTdjYTZjNzc3ZjVjYmYzZTFhYWRhNzRmMzQ4Yzk4NzlkNzhhIiwiJHNjaGVtYSI6Imh0dHA6Ly9raWx0LXByb3RvY29sLm9yZy9kcmFmdC0wMS9jdHlwZSMiLCJ0aXRsZSI6IkJMLU1haWwtU2ltcGxlIiwicHJvcGVydGllcyI6eyJGdWxsIG5hbWUiOnsidHlwZSI6InN0cmluZyJ9LCJFbWFpbCI6eyJ0eXBlIjoic3RyaW5nIn19LCJ0eXBlIjoib2JqZWN0In1dLCJhdHRlc3Rlck5hbWUiOiJTb2NpYWxLWUMiLCJhdHRlc3RlckRpZCI6ImRpZDpraWx0OjRvZUo3Nmhkajg0eG53Q05xaWpVSFVDVG1md1hnU1o0dm14TEVpVEVZZ1FkQkNjWiIsInNwZWNWZXJzaW9uIjoiMy4wIn0="
        title="Light DID Identity"
      />
      <ul
        class="bubbles"
      >
        <li
          class="item"
        >
          <a
            aria-current="page"
            aria-label="Light DID Identity"
            class="bubble bubbleActive"
            href="/?data=eyJjbGFpbSI6eyJjVHlwZUhhc2giOiIweDI0MGM3NDQ5MjNiZDk4Nzk3NTA0MzI4ZjliZWY1N2NhNmM3NzdmNWNiZjNlMWFhZGE3NGYzNDhjOTg3OWQ3OGEiLCJjb250ZW50cyI6eyJGdWxsIE5hbWUiOiJJbmdvIFL8YmUiLCJFbWFpbCI6ImluZ29Aa2lsdC5pbyJ9LCJvd25lciI6ImRpZDpraWx0OmxpZ2h0OjAwNG9VaUszRXZUY3pjNnVrWUxFVjU3QnV4emtXOXprc0tSY2haWFlHU29jTG1XRTVOIn0sImxlZ2l0aW1hdGlvbnMiOltdLCJxdW90ZSI6eyJhdHRlc3RlclNpZ25hdHVyZSI6eyJrZXlVcmkiOiJkaWQ6a2lsdDo0b2VKNzZoZGo4NHhud0NOcWlqVUhVQ1RtZndYZ1NaNHZteExFaVRFWWdRZEJDY1ojMHg1OWE1NmUxYTQwMTdiNjBhMDBjNTQ1MGJmZDZhZjIxZGM3NTNlODRlZjcyZjJmZDQ1ZGZjMjdkZDVjM2JkYWZiIiwic2lnbmF0dXJlIjoiMHgzMGQwMWFlZmQ2NGM3ZWVhNDhiYzc0Mzc4MmZlNTRiYTU4Zjk2YjVhOTFlMzNkOGQ5ZWU1ZmFlMjgyOTY0OTY3YmRlYTVlOWFjM2YzMTc1NmZkMjUxYTQwZWE0ZDJlOWE5M2FjOWIxMzE0Y2MwMzdjZGNiMDRhNTNhMDA3ZTY4ZiJ9LCJhdHRlc3RlckRpZCI6ImRpZDpraWx0OjRvZUo3Nmhkajg0eG53Q05xaWpVSFVDVG1md1hnU1o0dm14TEVpVEVZZ1FkQkNjWiIsImNUeXBlSGFzaCI6IjB4YmZhZDYwOTc3YmMxOGNmOWRmZDc2ZGE4ODYyNGNlMjE5MzYxZjMzN2I0MzMyZDVjNDJjMDQ3NDk5ZjRiOTNjNyIsImNvc3QiOnsiZ3Jvc3MiOjIzMywibmV0IjoyMy4zLCJ0YXgiOnsidmF0IjozLjN9fSwiY3VycmVuY3kiOiJLSUxUIiwidGltZWZyYW1lIjoiU2F0IEp1bCAxMCAyMDIxIDAwOjAwOjAwIEdNVCswMDAwIChDb29yZGluYXRlZCBVbml2ZXJzYWwgVGltZSkiLCJ0ZXJtc0FuZENvbmRpdGlvbnMiOiJodHRwczovL3d3dy5leGFtcGxlLmNvbS90ZXJtcyJ9LCJjVHlwZXMiOlt7IiRpZCI6ImtpbHQ6Y3R5cGU6MHgyNDBjNzQ0OTIzYmQ5ODc5NzUwNDMyOGY5YmVmNTdjYTZjNzc3ZjVjYmYzZTFhYWRhNzRmMzQ4Yzk4NzlkNzhhIiwiJHNjaGVtYSI6Imh0dHA6Ly9raWx0LXByb3RvY29sLm9yZy9kcmFmdC0wMS9jdHlwZSMiLCJ0aXRsZSI6IkJMLU1haWwtU2ltcGxlIiwicHJvcGVydGllcyI6eyJGdWxsIG5hbWUiOnsidHlwZSI6InN0cmluZyJ9LCJFbWFpbCI6eyJ0eXBlIjoic3RyaW5nIn19LCJ0eXBlIjoib2JqZWN0In1dLCJhdHRlc3Rlck5hbWUiOiJTb2NpYWxLWUMiLCJhdHRlc3RlckRpZCI6ImRpZDpraWx0OjRvZUo3Nmhkajg0eG53Q05xaWpVSFVDVG1md1hnU1o0dm14TEVpVEVZZ1FkQkNjWiIsInNwZWNWZXJzaW9uIjoiMy4wIn0="
            title="Light DID Identity"
          />
        </li>
        <li
          class="item"
        >
          <a
            aria-current="page"
            aria-label="Identity with on-chain DID and lots of coins"
            class="bubble bubbleActive"
            href="/?data=eyJjbGFpbSI6eyJjVHlwZUhhc2giOiIweDI0MGM3NDQ5MjNiZDk4Nzk3NTA0MzI4ZjliZWY1N2NhNmM3NzdmNWNiZjNlMWFhZGE3NGYzNDhjOTg3OWQ3OGEiLCJjb250ZW50cyI6eyJGdWxsIE5hbWUiOiJJbmdvIFL8YmUiLCJFbWFpbCI6ImluZ29Aa2lsdC5pbyJ9LCJvd25lciI6ImRpZDpraWx0OmxpZ2h0OjAwNG9VaUszRXZUY3pjNnVrWUxFVjU3QnV4emtXOXprc0tSY2haWFlHU29jTG1XRTVOIn0sImxlZ2l0aW1hdGlvbnMiOltdLCJxdW90ZSI6eyJhdHRlc3RlclNpZ25hdHVyZSI6eyJrZXlVcmkiOiJkaWQ6a2lsdDo0b2VKNzZoZGo4NHhud0NOcWlqVUhVQ1RtZndYZ1NaNHZteExFaVRFWWdRZEJDY1ojMHg1OWE1NmUxYTQwMTdiNjBhMDBjNTQ1MGJmZDZhZjIxZGM3NTNlODRlZjcyZjJmZDQ1ZGZjMjdkZDVjM2JkYWZiIiwic2lnbmF0dXJlIjoiMHgzMGQwMWFlZmQ2NGM3ZWVhNDhiYzc0Mzc4MmZlNTRiYTU4Zjk2YjVhOTFlMzNkOGQ5ZWU1ZmFlMjgyOTY0OTY3YmRlYTVlOWFjM2YzMTc1NmZkMjUxYTQwZWE0ZDJlOWE5M2FjOWIxMzE0Y2MwMzdjZGNiMDRhNTNhMDA3ZTY4ZiJ9LCJhdHRlc3RlckRpZCI6ImRpZDpraWx0OjRvZUo3Nmhkajg0eG53Q05xaWpVSFVDVG1md1hnU1o0dm14TEVpVEVZZ1FkQkNjWiIsImNUeXBlSGFzaCI6IjB4YmZhZDYwOTc3YmMxOGNmOWRmZDc2ZGE4ODYyNGNlMjE5MzYxZjMzN2I0MzMyZDVjNDJjMDQ3NDk5ZjRiOTNjNyIsImNvc3QiOnsiZ3Jvc3MiOjIzMywibmV0IjoyMy4zLCJ0YXgiOnsidmF0IjozLjN9fSwiY3VycmVuY3kiOiJLSUxUIiwidGltZWZyYW1lIjoiU2F0IEp1bCAxMCAyMDIxIDAwOjAwOjAwIEdNVCswMDAwIChDb29yZGluYXRlZCBVbml2ZXJzYWwgVGltZSkiLCJ0ZXJtc0FuZENvbmRpdGlvbnMiOiJodHRwczovL3d3dy5leGFtcGxlLmNvbS90ZXJtcyJ9LCJjVHlwZXMiOlt7IiRpZCI6ImtpbHQ6Y3R5cGU6MHgyNDBjNzQ0OTIzYmQ5ODc5NzUwNDMyOGY5YmVmNTdjYTZjNzc3ZjVjYmYzZTFhYWRhNzRmMzQ4Yzk4NzlkNzhhIiwiJHNjaGVtYSI6Imh0dHA6Ly9raWx0LXByb3RvY29sLm9yZy9kcmFmdC0wMS9jdHlwZSMiLCJ0aXRsZSI6IkJMLU1haWwtU2ltcGxlIiwicHJvcGVydGllcyI6eyJGdWxsIG5hbWUiOnsidHlwZSI6InN0cmluZyJ9LCJFbWFpbCI6eyJ0eXBlIjoic3RyaW5nIn19LCJ0eXBlIjoib2JqZWN0In1dLCJhdHRlc3Rlck5hbWUiOiJTb2NpYWxLWUMiLCJhdHRlc3RlckRpZCI6ImRpZDpraWx0OjRvZUo3Nmhkajg0eG53Q05xaWpVSFVDVG1md1hnU1o0dm14TEVpVEVZZ1FkQkNjWiIsInNwZWNWZXJzaW9uIjoiMy4wIn0="
            title="Identity with on-chain DID and lots of coins"
          />
        </li>
        <li
          class="item"
        >
          <a
            aria-current="page"
            aria-label="Identity which has a very very very very long name and might not even fit on the screen"
            class="bubble bubbleActive"
            href="/?data=eyJjbGFpbSI6eyJjVHlwZUhhc2giOiIweDI0MGM3NDQ5MjNiZDk4Nzk3NTA0MzI4ZjliZWY1N2NhNmM3NzdmNWNiZjNlMWFhZGE3NGYzNDhjOTg3OWQ3OGEiLCJjb250ZW50cyI6eyJGdWxsIE5hbWUiOiJJbmdvIFL8YmUiLCJFbWFpbCI6ImluZ29Aa2lsdC5pbyJ9LCJvd25lciI6ImRpZDpraWx0OmxpZ2h0OjAwNG9VaUszRXZUY3pjNnVrWUxFVjU3QnV4emtXOXprc0tSY2haWFlHU29jTG1XRTVOIn0sImxlZ2l0aW1hdGlvbnMiOltdLCJxdW90ZSI6eyJhdHRlc3RlclNpZ25hdHVyZSI6eyJrZXlVcmkiOiJkaWQ6a2lsdDo0b2VKNzZoZGo4NHhud0NOcWlqVUhVQ1RtZndYZ1NaNHZteExFaVRFWWdRZEJDY1ojMHg1OWE1NmUxYTQwMTdiNjBhMDBjNTQ1MGJmZDZhZjIxZGM3NTNlODRlZjcyZjJmZDQ1ZGZjMjdkZDVjM2JkYWZiIiwic2lnbmF0dXJlIjoiMHgzMGQwMWFlZmQ2NGM3ZWVhNDhiYzc0Mzc4MmZlNTRiYTU4Zjk2YjVhOTFlMzNkOGQ5ZWU1ZmFlMjgyOTY0OTY3YmRlYTVlOWFjM2YzMTc1NmZkMjUxYTQwZWE0ZDJlOWE5M2FjOWIxMzE0Y2MwMzdjZGNiMDRhNTNhMDA3ZTY4ZiJ9LCJhdHRlc3RlckRpZCI6ImRpZDpraWx0OjRvZUo3Nmhkajg0eG53Q05xaWpVSFVDVG1md1hnU1o0dm14TEVpVEVZZ1FkQkNjWiIsImNUeXBlSGFzaCI6IjB4YmZhZDYwOTc3YmMxOGNmOWRmZDc2ZGE4ODYyNGNlMjE5MzYxZjMzN2I0MzMyZDVjNDJjMDQ3NDk5ZjRiOTNjNyIsImNvc3QiOnsiZ3Jvc3MiOjIzMywibmV0IjoyMy4zLCJ0YXgiOnsidmF0IjozLjN9fSwiY3VycmVuY3kiOiJLSUxUIiwidGltZWZyYW1lIjoiU2F0IEp1bCAxMCAyMDIxIDAwOjAwOjAwIEdNVCswMDAwIChDb29yZGluYXRlZCBVbml2ZXJzYWwgVGltZSkiLCJ0ZXJtc0FuZENvbmRpdGlvbnMiOiJodHRwczovL3d3dy5leGFtcGxlLmNvbS90ZXJtcyJ9LCJjVHlwZXMiOlt7IiRpZCI6ImtpbHQ6Y3R5cGU6MHgyNDBjNzQ0OTIzYmQ5ODc5NzUwNDMyOGY5YmVmNTdjYTZjNzc3ZjVjYmYzZTFhYWRhNzRmMzQ4Yzk4NzlkNzhhIiwiJHNjaGVtYSI6Imh0dHA6Ly9raWx0LXByb3RvY29sLm9yZy9kcmFmdC0wMS9jdHlwZSMiLCJ0aXRsZSI6IkJMLU1haWwtU2ltcGxlIiwicHJvcGVydGllcyI6eyJGdWxsIG5hbWUiOnsidHlwZSI6InN0cmluZyJ9LCJFbWFpbCI6eyJ0eXBlIjoic3RyaW5nIn19LCJ0eXBlIjoib2JqZWN0In1dLCJhdHRlc3Rlck5hbWUiOiJTb2NpYWxLWUMiLCJhdHRlc3RlckRpZCI6ImRpZDpraWx0OjRvZUo3Nmhkajg0eG53Q05xaWpVSFVDVG1md1hnU1o0dm14TEVpVEVZZ1FkQkNjWiIsInNwZWNWZXJzaW9uIjoiMy4wIn0="
            title="Identity which has a very very very very long name and might not even fit on the screen"
          />
        </li>
        <li
          class="item"
        >
          <a
            aria-current="page"
            aria-label="Add Identity"
            class="add addActive"
            href="/"
            title="Add Identity"
          />
        </li>
      </ul>
    </div>
    <p
      class="costs"
    >
      <span>
        Credential Costs: 
      </span>
      <span
        class="amount"
      >
        233
        <span
          class="fractional"
        >
          .0000
        </span>
         
        <span
          aria-label="KILT Coin"
          class="coin"
        />
      </span>
    </p>
    <dl
      class="details"
    >
      <dt
        class="detailName"
      >
        Full Name
        :
      </dt>
      <dd
        class="detailValue"
      >
        Ingo Rï¿½be
      </dd>
      <dt
        class="detailName"
      >
        Email
        :
      </dt>
      <dd
        class="detailValue"
      >
        ingo@kilt.io
      </dd>
      <dt
        class="detailName"
      >
        Credential type
        :
      </dt>
      <dd
        class="detailValue"
      >
        BL-Mail-Simple
      </dd>
      <dt
        class="detailName"
      >
        Attester
        :
      </dt>
      <dd
        class="detailValue"
      >
        SocialKYC
      </dd>
    </dl>
    <p
      class="resetLine"
    >
      <label
        class="passwordLabel"
        for="password"
      >
        Sign with password:
      </label>
      <a
        class="reset"
        href="/identity/4rZ7pGtvmLhAYesf7DAzLQixdTEwWPN3emKb44bKVXqSoTZB/reset"
      >
        Reset password
      </a>
    </p>
    <p
      class="passwordLine"
    >
      <input
        class="password"
        id="password"
        name="password"
        type="password"
        value=""
      />
      <button
        aria-label="Show"
        class="show"
        title="Show"
        type="button"
      />
      <output
        class="errorTooltip"
        hidden=""
      />
    </p>
    <label
      class="rememberLabel"
    >
      <span>
        Remember password for the next 15 minutes
      </span>
      <input
        class="remember"
        name="remember"
        type="checkbox"
      />
      <span />
    </label>
    <p
      class="buttonsLine"
    >
      <button
        class="cancel"
        type="button"
      >
        Cancel
      </button>
      <button
        class="submit"
        disabled=""
        type="submit"
      >
        Request Attestation
      </button>
      <output
        class="errorTooltip"
      >
        DID deleted, please select a different Identity
      </output>
    </p>
  </form>
</div>
`;
